alws/crud/release.py

```diff
@@ -147,33 +167,40 @@ def get_pulp_based_response():
     }
 
     added_packages = []
-    if not settings.package_beholder_enabled:
-        return get_pulp_based_response()
 
-    clean_ref_dist_name = re.search(
-        r'(?P<dist_name>[a-z]+)', reference_dist_name,
-        re.IGNORECASE).groupdict().get('dist_name')
-    clean_ref_dist_name_lower = clean_ref_dist_name.lower()
-    beholder = BeholderClient(settings.beholder_host)
     for module in pulp_rpm_modules:
         endpoint = (
             f'/api/v1/distros/{clean_ref_dist_name}/'
             f'{reference_dist_version}/module/{module["name"]}/'
             f'{module["stream"]}/{module["arch"]}/'
         )
-        module_response = await beholder.get(endpoint)
-        module_repo = module_response['repository']
-        repo_name = repo_name_regex.search(
-            module_repo['name']).groupdict()['name']
-        release_repo_name = (f'{clean_ref_dist_name_lower}'
-                             f'-{base_dist_version}-{repo_name}')
-        module_info = {
-            'module': module,
-            'repositories': [
-                RepoType(release_repo_name, module_repo['arch'], False)
-            ]
-        }
-        rpm_modules.append(module_info)
+        success = True
+        module_response = None
+        try:
+            module_response = await beholder.get(endpoint)
+        except Exception:
+            endpoint = (
+                f'/api/v1/distros/{base_dist_name}/'
+                f'{reference_dist_version}/module/{module["name"]}/'
+                f'{module["stream"]}/{module["arch"]}/'
+            )
+            try:
+                module_response = await beholder.get(endpoint)
+            except Exception:
+                success = False
```

Comment:
can be for loop with instead of manually handling exception two times
smth like this:
```
success = False
for endpoint in [A, B]:
   try:
     module_response = await beholder.get(endpoint)
  except Exception:
    continue
  success = True
```